src/main/java/org/jitsi/srtp/crypto/Aes.java

```diff
@@ -872,27 +894,46 @@ public static synchronized Provider getProvider()
 
                     if (Provider.class.isAssignableFrom(clazz))
                     {
-                        Constructor<?> contructor
-                            = clazz.getConstructor(String.class);
-
-                        // The SunPKCS11 Config name should be unique in order
-                        // to avoid repeated initialization exceptions.
-                        String name = null;
-                        Package pkg = Aes.class.getPackage();
-
-                        if (pkg != null)
-                            name = pkg.getName();
-                        if (name == null || name.length() == 0)
-                            name = "org.jitsi.srtp";
-
-                        provider
-                            = (Provider)
-                                contructor.newInstance(
-                                        "--name=" + name + "\\n"
-                                            + "nssDbMode=noDb\\n"
-                                            + "attributes=compatibility");
+                        try
+                        {
+                            /* Java 9+.  PKCS11 configuration is set using the
+                             * configure method.
+                             * Use reflection so we can build with JDK 8. */
+                            Method configureMethod =
+                                Provider.class.getMethod("configure", String.class);
+
+                            Provider prototype =
+                                Security.getProvider("SunPKCS11");
+                            if (prototype != null)
+                            {
+                                provider
+                                    = (Provider)
+                                    configureMethod.invoke(prototype, getConfiguration());
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            logger.debug("Unable to construct PKCS11 provider: " + e.getMessage());
+                        }
+                        finally
```

Comment:
The entire finally block could just be the else-part of the provier-null check above and the inner try/catch made redundant.

Reply:
I'm not sure I follow? This is executed if `getMethod("configure")` throws, as it will in Java 1.8.